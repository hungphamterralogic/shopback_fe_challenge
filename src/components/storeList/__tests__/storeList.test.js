import React from 'react';

import { genInitialMockStore, renderWithStore } from 'test/reduxComponentTestHelpers';

import detectDevice from 'utils/detectDevice';

import StoreList from '../storeList';

const stores = [
  { id: 'id01', logoUrl: 'logoUrl01', displayText: 'displayText01' },
  { id: 'id02', logoUrl: 'logoUrl02', displayText: 'displayText02' },
  { id: 'id03', logoUrl: 'logoUrl03', displayText: 'displayText03' },
  { id: 'id04', logoUrl: 'logoUrl04', displayText: 'displayText04' },
  { id: 'id05', logoUrl: 'logoUrl05', displayText: 'displayText05' },
  { id: 'id06', logoUrl: 'logoUrl06', displayText: 'displayText06' },
  { id: 'id07', logoUrl: 'logoUrl07', displayText: 'displayText07' },
  { id: 'id08', logoUrl: 'logoUrl08', displayText: 'displayText08' },
  { id: 'id09', logoUrl: 'logoUrl09', displayText: 'displayText09' },
  { id: 'id10', logoUrl: 'logoUrl10', displayText: 'displayText10' },
  { id: 'id11', logoUrl: 'logoUrl11', displayText: 'displayText11' },
  { id: 'id12', logoUrl: 'logoUrl12', displayText: 'displayText12' }
];

describe('StoreList', () => {
  describe('display properly', () => {
    test('is mobile', () => {
      const isMobileMock = jest.spyOn(detectDevice, 'isMobile');
      isMobileMock.mockImplementation(() => true);

      const reduxStore = genInitialMockStore();
      const renderer = renderWithStore(<StoreList stores={stores} />, reduxStore);

      expect(renderer.toJSON()).toMatchSnapshot();
      isMobileMock.mockRestore();
    });

    test('is not mobile', () => {
      const isMobileMock = jest.spyOn(detectDevice, 'isMobile');
      isMobileMock.mockImplementation(() => false);

      const reduxStore = genInitialMockStore();
      const renderer = renderWithStore(<StoreList stores={stores} />, reduxStore);

      expect(renderer.toJSON()).toMatchSnapshot();
      isMobileMock.mockRestore();
    });
  });
});
