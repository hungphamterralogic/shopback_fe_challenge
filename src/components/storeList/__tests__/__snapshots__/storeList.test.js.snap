// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreList display properly is mobile 1`] = `
<div
  className="StoreList-storeList-0-1-1"
>
  <div
    className="StoreItem-storeItem-0-1-2 StoreItem-storeItemMobile-0-1-3"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl01"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText01
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2 StoreItem-storeItemMobile-0-1-3"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl02"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText02
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2 StoreItem-storeItemMobile-0-1-3"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl03"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText03
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2 StoreItem-storeItemMobile-0-1-3"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl04"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText04
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2 StoreItem-storeItemMobile-0-1-3"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl05"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText05
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2 StoreItem-storeItemMobile-0-1-3"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl06"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText06
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2 StoreItem-storeItemMobile-0-1-3"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl07"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText07
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2 StoreItem-storeItemMobile-0-1-3"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl08"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText08
    </span>
  </div>
  <div
    className="ShowMore-showMoreMobile-0-1-7"
  >
    <div
      className="ShowMore-textBoxMobile-0-1-9"
    >
      All Stores
    </div>
  </div>
</div>
`;

exports[`StoreList display properly is not mobile 1`] = `
<div
  className="StoreList-storeList-0-1-1"
>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl01"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText01
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl02"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText02
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl03"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText03
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl04"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText04
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl05"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText05
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl06"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText06
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl07"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText07
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl08"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText08
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl09"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText09
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl10"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText10
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl11"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText11
    </span>
  </div>
  <div
    className="StoreItem-storeItem-0-1-2"
    onClick={[Function]}
  >
    <img
      alt="icon"
      className="StoreItem-icon-0-1-4"
      src="logoUrl12"
    />
    <span
      className="StoreItem-name-0-1-5"
    >
      displayText12
    </span>
  </div>
  <div
    className="ShowMore-showMoreDesktop-0-1-6"
  >
    <div
      className="ShowMore-textBoxDesktop-0-1-8"
    >
      view all popular stores
    </div>
  </div>
</div>
`;
